sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java

```diff
@@ -73,66 +72,70 @@
 
         _configuration = configuration;
         _accessor = createAccessor(configuration, client, metricsRegistry);
-        _executorService = Executors.newFixedThreadPool(_configuration.getConcurrency());
+        _executorService = Executors.newFixedThreadPool(configuration.getConcurrency());
+        _successCount = new AtomicLong(0);
+        _errorCount = new AtomicLong(0);
+        _semaphore = new Semaphore(configuration.getConcurrency());
     }
 
-    public void run(Map<Key, ObjectNode> testData) {
-        final ArrayList<Key> keys = new ArrayList<>(testData.keySet());
+    public void run(final Set<Key> testKeys) {
+        final ArrayList<Key> keys = new ArrayList<>(testKeys);
         Collections.shuffle(keys);
         final long runStartTime = System.currentTimeMillis();
-        final AtomicLong successCount = new AtomicLong(0);
-        final AtomicLong errorCount = new AtomicLong(0);
         long i = 0;
         for (; BenchmarkHelper.shouldContinue(runStartTime, i, _configuration); i++) {
+            if (exceedsErrorThreshold()) {
```

Comment:
What is the motivation here ? We should not stop the run on fixed error count, instead let the run complete and analysis the result later

Reply:
After a few runs, I realize it's not adding much value. The error stats are counting and reported separately, so removing it.